out/test/addressbook-level2/input.txt

```diff
@@ -0,0 +1,167 @@
+  delete 1
+  view 1
+  viewall 1
+
+##########################################################
+# clean and check state
+##########################################################
+
+  # address book should be emptied
+  clear
+  list
+
+##########################################################
+# test add person command, setup state for further tests
+##########################################################
+
+  # should catch invalid args format
+  add wrong args wrong args
+  add Valid Name 12345 e/valid@email.butNoPhonePrefix a/valid, address
+  add Valid Name p/12345 valid@email.butNoPrefix a/valid, address
+  add Valid Name p/12345 e/valid@email.butNoAddressPrefix valid, address
+  add Valid Name p/12345 e/valid@email.butNoTagPrefix a/valid, address t/goodTag noPrefixTag
+  
+  # should catch invalid person data
+  add []\[;] p/12345 e/valid@e.mail a/valid, address
+  add Valid Name p/not_numbers e/valid@e.mail a/valid, address
+  add Valid Name p/12345 e/notAnEmail a/valid, address
+  add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag
+
+  # should add correctly and list non private information
+  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street
+  list
+  add Betsy Choo pp/222222 pe/benchoo@nus.edu.sg pa/222, beta street t/secretive
+  list
+  add Charlie Dickson pp/333333 e/charlie.d@nus.edu.sg a/333, gamma street t/friends t/school
+  list
+  add Dickson Ee p/444444 pe/dickson@nus.edu.sg a/444, delta street t/friends
+  list
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+  list
+
+  # should not allow adding duplicate persons
+  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy
+
+##########################################################
+# test view/viewall persons command
+##########################################################
+
+  # should catch invalid args format
+  view
+  viewall
+  view should be only one number
+  viewall should only be one number
+
+  # should catch invalid index
+  view -1
+  view 0
+  view 6
+  viewall -1
+  viewall 0
+  viewall 6
+
+  # should show all non private for both view and viewall
+  view 1
+  viewall 1
+
+  # should show only non private details for view
+  view 3
+  view 4
+  view 5
+
+  # should show all details even private for viewall
+  viewall 3
+  viewall 4
+  viewall 5
+
+##########################################################
+# test find persons command
+##########################################################
+
+  # should consider no keywords as invalid command format
+  find
+  # should only match full words in person names
+  find bet
+  # does not match if none have keyword
+  find 23912039120
+  # matching should be case-sensitive
+  find betsy
+
+  # find unique keyword
+  find Betsy
+  # find multiple with same keyword
+  find Dickson
+  # find multiple with some keywords
+  find Charlie Betsy
+  # find suggestions with single letter
```

Comment:
Nice work on updating your tests (: Instead of replacing the entire input.txt file, try to limit your commit changes to the feature you are implementing (i.e. just add and commit the relevant changes)